modules/fatfs/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+if(CONFIG_FAT_FILESYSTEM_ELM)
+  add_library(ELMFAT INTERFACE)
```

Comment:
a comment here as to why a `ELMFAT` interface library is needed.

Who uses this interface lib, and is it in fact needed ?

Reply:
It is user here:
https://github.com/zephyrproject-rtos/zephyr/blob/4bae4f426ff4c5e4b84f29d8150789db1c7c569b/subsys/fs/CMakeLists.txt#L19